doc/website/getting-started/overview.md

```diff
@@ -0,0 +1,540 @@
+cases but inefficient in general, as it often leads to unnecessary latency and wake-ups without data.
+
+The ``Waitset`` can be used to relinquish control (putting the thread to sleep) and wait for user defined conditions
+to become true. Usually these conditions correspond to the availability of data at specific subscribers. This way we
+can (almost) immediately wake up when data is available and will avoid unnecessary wake-ups if no data is available.
+
+To do so it manages a set of triggers which can be activated and indicate that a corresponding condition became true
+which in turn will wake up a potentially waiting thread. In this way it extends a condition variable to a collection
+of conditions. Upon waking up it can be determined which conditions became true and caused the wake up. In the case
+that the wake up event was the availability of new data, this data can now be collected at the subscriber.
+
+For more information on how to use the Waitset see [Waitset](../../../iceoryx_examples/waitset/README.md).
+
+## API
+
+Now, we show how the API can be used to establish a publish-subscribe communication in an iceoryx system. Many parts
+of the API follow a functional programming approach and allow the user to specify functions which handle the possible
+cases, e.g. what should happen when data is received.
+
+This is very flexible but requires using the monadic types ``cxx::expected`` and ``cxx::optional``, which we
+introduce in the following sections.
+
+We distinguish between the ``typed API`` and the ``untyped API``. In the Typed API the underlying data type is made
+apparent by typed pointers or references to some data type T (often a template parameter). this allows working with
+the data in an C++ idiomatic and type-safe way and should be preferred whenever possible.
+
+The Untyped API provides opaque (i.e. void) pointers to data, which is flexible and efficient but also requires that
+the user takes care to interpret received data correctly, i.e. as a type compatible to what was actually sent. This
+is required for interaction with other lower level APIs and should be used sparingly. For further information see the
+respective header files.
+
+There also is a plain [C API](../../../iceoryx_examples/icedelivery_on_c/README.md), which can be used if C++ is not
+an option.
+
+We now describe the how to use the API in iceoryx applications. We will omit namespaces in several places to keep
+the code concise. In most cases it can be assumed that we are using namespace ``iox::cxx``. We also will use ``auto``
+sparingly to clearly show which types are involved, but in many cases automatic type deduction is possible and can
+shorten the code.
+
+### Optional
+
+The type ``cxx::optional<T>`` is used to indicate that there may or may not be a value of a specific type ``T``
+available. This is essentially the maybe monad in functional programming. Assuming we have some optional (usually the
```

Comment:
Please add a link to wikipedai `[monad](https://en.wikipedia.org/wiki/Monad_(functional_programming))`

Reply:
Why? They are just monoids in the category of endofunctors...everyone knows that.

Yeah, can do.

Reply:
Totally agree, we should always try to educate newbies by pointing them the respective ressources